src/components/calcite-button/calcite-button.tsx

```diff
@@ -138,15 +138,22 @@ export class CalciteButton {
       </div>
     );
 
-    const icon = (
-      <svg
+    const iconScale =
```

Comment:
I sometimes find it easier to just make a lookup for things like this as the ternaries can get pretty complicated:

```
const iconScales = {
  xs: "s",
  s: "s",
  m: "s",
  l: "m"
}

iconScales[this.scale] || "l"
```